### ISSUE TITLE
Linen MultiHeadAttention applies constant and learnable alpha at different stages

### ISSUE THREAD

**Issue by mlnomadpy** [azettaai/nmn#52]
## Summary

`nmn.linen.MultiHeadAttention` applies a constant alpha to the already-normalized attention output, while a learnable alpha is passed into the YAT score calculation before softmax/L1 normalization. Therefore a constant alpha and a learnable alpha set to the same value implement different functions.

## Affected code

- `src/nmn/linen/attention.py`
- `MultiHeadAttention.__call__`

## Reproduction

On commit `4f0298d`, use identical projection parameters and set both alpha variants to 2:

```python
import jax
import jax.numpy as jnp
from flax.core import freeze, unfreeze
from nmn.linen.attention import MultiHeadAttention

x = jax.random.normal(jax.random.key(1), (1, 4, 8))
learnable = MultiHeadAttention(num_heads=2, use_alpha=True, constant_alpha=None)
constant = MultiHeadAttention(num_heads=2, use_alpha=True, constant_alpha=2.0)

variables = learnable.init(jax.random.key(0), x)
p = unfreeze(variables)
p["params"]["alpha"] = jnp.array([2.0])
learnable_variables = freeze(p)

p = unfreeze(learnable_variables)
del p["params"]["alpha"]
constant_variables = freeze(p)

y_learnable = learnable.apply(learnable_variables, x)
y_constant = constant.apply(constant_variables, x)
print(float(jnp.max(jnp.abs(y_learnable - y_constant))))
```

Observed maximum difference: `0.8976223`.

The learnable path calls `_nnx_yat_attention(..., alpha=alpha_val)`, which scales scores before normalization. The constant path calls it with `alpha=None` and then executes `x = x * _constant_alpha_value` after attention.

## Impact

Switching between fixed and trainable alpha changes attention semantics rather than only parameter trainability. Constant-alpha Linen attention also differs from the shared NNX score-scaling contract.

## Acceptance criteria

- [ ] Constant alpha is passed through the same score-scaling path as learnable alpha.
- [ ] Constant and learnable alpha produce matching outputs when their values and all projection parameters match.
- [ ] Tests cover softmax and L1 normalization, including alpha values other than 1.
- [ ] Documentation states the common alpha semantics.

**Comment by mlnomadpy** [comment]
The same semantic defect is present in two NNX modules and should be included in scope:

- `src/nmn/nnx/layers/attention/multi_head.py`
- `src/nmn/nnx/layers/attention/rotary_yat.py`

Both pass learnable alpha into the attention function (pre-normalization/logit scaling), but apply `constant_alpha` by multiplying the already-normalized attention output. A deterministic `alpha=sqrt(2)` reproduction differs by up to `1.38` between those two routes on a tiny Q/K/V example.

Suggested additional acceptance criteria:

- [ ] Align constant and learnable alpha placement in NNX `MultiHeadAttention` and `RotaryYatAttention`.
- [ ] Add equal-value constant-vs-learnable numerical parity tests for outputs and gradients.
